app/components/new-sign-up/form.hbs

```diff
@@ -7,14 +7,14 @@
   </div>
 
   <div class="signup-action__btn">
-    <SignUp::Button
-      @clickHandler={{if (eq @state "username")
-      @register @changeRouteParams}}
+    <NewSignUp::Button
+      @clickHandler={{@handleRouteParamsChange}}
       @state={{@state}}
       @disabled={{@isButtonDisabled}}
       @isSubmitClicked={{@isSubmitClicked}}
+      @routeParams={{@routeParams}}
```

Comment:
Why does the button need routeParams? Seems like code smell to me.

Reply:
re-factored the component. Found an even better way of handling this change. Thanks